src/selection-recommendation/time_conflicts_check.py

```diff
@@ -0,0 +1,60 @@
+def time_to_num(time):
+    """
+    time: a string representing a time, with hour and minute separated by a colon (:)
+    Returns a number
+    
+    e.g. time_to_num(9:00) -> 9
+         time_to_num(21:00) -> 21
+         time_to_num(12:30) -> 12.5
+    """
+    time_comps = time.split(":")
+    if int(time_comps[1]) == 0:
+        return int(time_comps[0])
+    return(int(time_comps[0]) + int(time_comps[1])/60)
+    
+
+def process_times(times):
+    """
+    times: a string of weekdays and time-slots of courses, delimited by spaces
+    Returns a list of tuples containing the weekday, start time, and duration of the courses
+    
+    e.g. process_times("MONDAY 18:00-20:00 THURSDAY 18:00-21:00") -> [("MONDAY", 18, 2), ("THURSDAY", 18, 3)]
+         process_times("WEDNESDAY 13:30-15:00") -> [("WEDNESDAY", 13.5, 1.5)]
+         
+    IMPORTANT: this function assumes that the input string is in a format similar to the examples above
+    """
+    all_times = []
+    times_comps = times.split(" ")
+    
+    for i in range(0, len(times_comps), 2):
+        start_end = times_comps[i + 1].split("-")
+        start = time_to_num(start_end[0])
+        dur = time_to_num(start_end[1]) - time_to_num(start_end[0])
+        all_times.append((times_comps[i], start, dur))
+        
+    return all_times
+    
+
+def no_time_conflict(times, times_to_check):
```

Comment:
I think I know what are you going to do with the code and I have a straightforward idea about how to eliminate conflict cases:

For every possible combination of timetable, let's make a 2-d array to store the information about that timetable, i.e., let A be a 14*5 zero matrix (14 hours everyday, 5 days every week). And we make a dictionary which map the combination to that matrix. Then based on the given course times, we add 1 to the time slot which is occupied. Finally, we pick all the valid timetable (which has no entry whose value is more than 1).

Then we can return all the possible combinations for further use.

I'm not sure if there are any better algorithm. If you think I misunderstand your algorithm and the algorithm is better than this, ignore all my words :p